apps/condo/domains/common/utils/serverSchema/changeTrackable.js

```diff
@@ -0,0 +1,355 @@
+ * `fields` object is a set of fields from schema, that is going to be change trackable,
+ * it's a "source" schema.
+ *
+ * Omitting of fields is not implemented here because trackable fields set
+ * should be used in different places and calling this function over and over
+ * is complicated
+ *
+ * @example
+ *
+ * // Schema for tracking changes of `Ticket`
+ * const TicketChange = new GQLListSchema('TicketChange', {
+ *      fields: {
+ *          // Something custom, e.g. reference to source schema,
+ *          // whose changes we will track in this schema
+ *          ticket: {
+ *              type: Relationship,
+ *              ref: 'Ticket',
+ *              isRequired: true,
+ *          },
+ *
+ *          // And this is an actual data-fields, that will store changes
+ *          // in the "Change storage set" format
+ *          ...generateChangeTrackableFieldsFrom(trackableFields)
+ *      }
+ * })
+ *
+ * @param {Object} fields - `fields` object of a Keystone schema
+ * @return {Object} - Set of fields, that should be substituted into a declaration of schema, that will store changes.
+ */
+function generateChangeTrackableFieldsFrom (fields) {
+    const scalars = _.transform(_.pickBy(fields, isScalar), mapScalars, {})
+    const relationsSingle = _.transform(_.pickBy(fields, isRelationSingle), mapRelationSingle, {})
+    const relationsMany = _.transform(_.pickBy(fields, isRelationMany), mapRelationMany, {})
+    return {
+        ...scalars,
+        ...relationsSingle,
+        ...relationsMany,
+    }
+}
+
+/**
+ * Map of field names to functions, that obtains item name by its id.
+ *
+ * @example
+ * const displayNameResolvers = {
+ *   'property': async (itemId) => {
+ *       const item = await getById('Property', itemId)
+ *       return _.get(item, 'name')
+ *   },
+ *   'status': async (itemId) => {
+ *       const item = await getById('TicketStatus', itemId)
+ *       return _.get(item, 'name')
+ *   },
+ *
+ * @typedef DisplayNameResolvers
+ */
+
+
+/**
+ * Map of fields with many-to-many relations to functions,
+ * that obtains ids and display names of previous and updated list of related items.
+ *
+ * @example
+ * // Resolver for `watchers` field of a schema with type `Relationship` and `many: true` option.
+ * const relatedManyToManyResolvers = {
+ *     'watchers': async ({ context, existingItem, originalInput }) => {
+ *          // Some logic, that finally returns following:
+ *          return {
+ *              existing: { ids, displayNames },
+ *              updated: { ids, displayNames },
+ *          }
+ *      }
+ * }
+ *
+ * @typedef RelatedManyToManyResolvers
+ */
+
+/**
+ * Should be substituted into `afterChange` hook of a schema, whose changes
+ * should be trackable.
+ *
+ * @param fields
+ * @param createCallback
+ * @param displayNameResolvers
+ * @param relatedManyToManyResolvers
+ * @return function, compatible with Keystone `afterChange` hook
+ */
+const afterChangeHook = (
```

Comment:
It's a too-common name for util function. And it's better to use a verb inside the function name